articles/is-node-js-a good-choice-to-create-an-app-based-on-micro-services/index.md

```diff
@@ -0,0 +1,36 @@
+
+In this article, the reader will understand the importance of using micro-services and the key benefits of micro-services when building apps. The reader will also understand the reason Node.js coupled with micro-services is such a good technology to implement when building complex apps and the benefits Node.js can offer during development.
+
+### What are micro-services?
+
+These are a software design pattern which splits up huge applications into a set of small and simple applications or services. The small chunks of applications or services can then be maintained with ease, tested, and [loosely coupled](https://nordicapis.com/the-difference-between-tight-coupling-and-loose-coupling/). This is followed by the independent deployment of these applications in an organized manner around their business capabilities.
+
+Earlier before the introduction of micro-services architecture, apps used to be built as [monoliths](http://www.codingthearchitecture.com/2014/11/19/what_is_a_monolith.html). They actually grew into a bigger size, causing problems with their maintainability. The logical solution was to split them up into smaller bits, known as micro-services.
+
+In a monolithic architecture, tasks are not clearly divided nor defined. For instance, fixing a login issue in a huge application could require a developer to first understand the whole codebase of an application, instead of just one straightforward process.
+
+#### Benefits of micro-services
+
+The choice of technology when building applications is not important compared to the business rationale. Introducing micro-services into an app development project provides tangible benefits, as highlighted below.
+
+1. Better structure: a well-structured app is usually easier to understand, meaning that any new features to be added to the app will be easier to design and implement. And therefore, less time and cost is spent on business analysis. Thus developers spend more time on development.
+2. Seamless implementation of complex products: complex applications are difficult to conceptualize, build, and maintain. But with micro-services, they are broken down into small bits that are more manageable.
+3. Scalability: scaling a monolithic app with a huge [blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob) of tightly coupled code is difficult, since a lot of time is spent trying to figure out what&#39;s going on inside. With micro-services, it becomes more apparent, which are the bottlenecks, and therefore getting them to handle more load is easier.
+4. Fewer errors: in a monolithic architecture looking for bugs is challenging and time consuming compared to micro-services where the small bit-sized pieces of technology are kept error-free and thus more manageable.
```

Comment:
byte-sized.

It would make more sense if you compared the larger size vs small size of monolithic vs micro-service in terms of bug-hunting. Just because a micro-service is small doesn't mean it's error-free.